src/shogun/machine/BaggingMachine.cpp

```diff
@@ -294,16 +269,16 @@ std::shared_ptr<CombinationRule> BaggingMachine::get_combination_rule() const
 	return m_combination_rule;
 }
 
-float64_t BaggingMachine::get_oob_error() const
+float64_t BaggingMachine::get_oob_error(const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& labs) const
```

Comment:
@karlnapf I think this is going to cause issues, because we use the state to get the out of bag error, and it is the only way to expose this to the interfaces. Do you think this should become a free function instead? Or maybe this should be made into a `Evaluation` subclass?

Reply:
maybe we can write something like this:
``` cpp
std::function<float64_t()> get_oob_error;
```
and in `train_machine`,
``` cpp
get_oob_error = [&](){return get_oob_error(data, labs)};
```
but I am not sure it is a good idea.

Reply:
The problem is that we have a hack to make this work with interfaces. We have the `.run("get_oob_error")`, because there is no way of accessing the specialised class, we only expose Machine. Not sure how we would pass data and labels with your suggestion?

Reply:
`get_oob_error` is like a function without parameter now, I guess we can just do previous way `watch_method(kOobError, &BaggingMachine::get_oob_error);`?

Reply:
yes, if it has no parameters, this is easy :)

Reply:
@LiuYuHui but the previous way relies on the state of BaggingMachine, which is something you are removing right?

Reply:
yes, now lambda function `get_oob_error ` capture the feature and the labels, so we don't have to pass the parameter.